climada/engine/impact_calc.py

```diff
@@ -129,6 +129,19 @@ def impact(self, save_mat=True, assign_centroids=True,
                 )
 
         impf_col = self.exposures.get_impf_column(self.hazard.haz_type)
+
+        # check for compability of impact function id between impact function set and exposure
+        if any(impf_id not in self.impfset.get_ids(haz_type=self.hazard.haz_type) for
+               impf_id in np.unique(self.exposures.gdf[impf_col].values)):
+            raise AttributeError(
+                "At least one impact function associated to the exposures has no match "
+                "in the impact function set.\nThe impact functions in the impact function"
+                f" set for hazard type \"{self.hazard.haz_type}\" have ids "
+                f"{self.impfset.get_ids(haz_type=self.hazard.haz_type)}. The column "
+                f"\"{impf_col}\" in the exposures contains the ids "
+                f"{np.unique(self.exposures.gdf[impf_col].values).astype(int).tolist()}.\n"
+                "Please make sure that all exposure points are associated with an impact "
+                "function that is included in the impact function set.")
```

Comment:
while we're at it why not just list the felonies:
```suggestion
                unknown_impact_functions = list(self.exposures.gdf[
                    ~self.exposures.gdf[impf_col].isin(known_impact_functions)
                ][impf_col].drop_duplicates().astype(int))
                raise ValueError(
                    f"The associated impact function(s) {', '.join(unknown_impact_functions)}"
                    " have no match in impact function set for hazard type '{self.hazard.haz_type}'\n"
                "Please make sure that all exposure points are associated with an impact "
                "function that is included in the impact function set.")
```

Reply:
Great, that's even better this way! I have included this modification in my last commits.